plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts

```diff
@@ -111,6 +120,47 @@ export class DefaultLocationStore implements LocationStore, EntityProvider {
     });
   }
 
+  async getLocationByEntity(entityRef: CompoundEntityRef): Promise<Location> {
+    const entityRefString = stringifyEntityRef(entityRef);
+
+    const [entity] = await this.db<DbRefreshStateRow>('refresh_state')
+      .where({ entity_ref: entityRefString })
+      .select('entity_id')
+      .limit(1);
+    if (!entity) {
+      throw new NotFoundError(`found no entity for ref ${entityRefString}`);
+    }
+
+    const [locationKeyValue] = await this.db<DbSearchRow>('search')
+      .where({
+        entity_id: entity.entity_id,
+        key: `metadata.annotations.${ANNOTATION_ORIGIN_LOCATION}`,
+      })
+      .select('value')
+      .limit(1);
+    if (!locationKeyValue) {
+      throw new NotFoundError(
+        `found no origin annotation for ref ${entityRefString}`,
+      );
+    }
+    const [type, ...rest] = locationKeyValue.value?.split(':') ?? [];
+    const target = rest.join(':');
```

Comment:
Might be worth using `parseLocationRef` to get a proper error if this ends up being a location ref of a form that we don't know how to parse. Not that that really is supposed to be able to happen 😅. In particular if the location doesn't contain a `:` the rest of this might get a bit strange.

Reply:
thanks, was unfamiliar with that helper